manifests/init.pp

```diff
@@ -254,6 +254,10 @@
 
   if $enable_docker {
     include pulpcore::plugin::container
+    include foreman_proxy_content::container
+    class { 'foreman_proxy::plugin::container_gateway':
+      pulp_endpoint => "https://${servername}",
+    }
```

Comment:
This is included unconditionally. Is it intended that it's also included when it's shared with a Foreman install?

Reply:
I put this here because I thought the container gateway should always be paired with `pulpcore::plugin::container` on a proxy.  I was assuming if Pulpcore is installed then Katello is in the picture somewhere too.  Is that not the case?

Reply:
Naming here is confusing, but we have 2 scenarios:
* Deployed standalone, typically a mirror. In downstream terms this would be a capsule
* Deployed together with Foreman

I think you handle the standalone case, but I'm asking about the latter case. From conversations I seem to recall that the container gateway plugin is only relevant in case there is no Katello serving the API.

You probably also create a nice loop where the container gateway will end up asking itself since `/v2` is proxied to it.

Reply:
Ah right, that slipped my mind. Right now I only want the container gateway to exist in mirrors. Katello handles the registry if the smart proxy exists on the same machine.

Is there a best parameter to check for a mirror? I believe in the pulp plugin "mirror" will be set to true. Perhaps there's a more general parameter?

Reply:
There is `$shared_with_foreman_vhost`.